src/asset.ts

```diff
@@ -0,0 +1,169 @@
+import algosdk, { Account, Algodv2 } from 'algosdk'
+import { Config, sendGroupOfTransactions } from '.'
+import { TransactionGroupToSend, TransactionToSign } from './types/transaction'
+
+const MaxTxGroupSize = 16
+
+function* chunks<T>(arr: T[], n: number): Generator<T[], void> {
+  for (let i = 0; i < arr.length; i += n) yield arr.slice(i, i + n)
+}
+
+async function ensureAccountIsValid(client: algosdk.Algodv2, account: algosdk.Account) {
+  try {
+    await client.accountInformation(account.addr).do()
+  } catch (error) {
+    throw new Error(`Account address ${account.addr} does not exist`)
+  }
+}
+
+async function ensureAssetFirstOptIn(client: algosdk.Algodv2, account: algosdk.Account, assetIds: number[]) {
```

Comment:
This method and the one below are redundant, both are almost identical except for min balance parameters, you can combine them into a single method called ensureAssetBalance and take a third param of custom enum type call ValidationType (with option OptIn or OptOut), based on this enum you can them figure out whether to check if balance is >= 0 (opted in) or asset should not be inside accountInfo response at all (opted out)